src/trio/_core/_run.py

```diff
@@ -883,7 +912,7 @@ def shield(self, new_value: bool) -> None:
             self._cancel_status.recalculate()
 
     @enable_ki_protection
-    def cancel(self) -> None:
+    def cancel(self, *, reason: str | None = None) -> None:
         """Cancels this scope immediately.
```

Comment:
I'd allow posargs here, since there's only one argument.  Also needs `reason=` to be explained in the docstring, e.g.

> The optional ``reason`` argument accepts a string, which will be attached to any resulting ``Cancelled`` exception, to help you understand where that cancellation is coming from and why it happened.